sled-agent/src/http_entrypoints.rs

```diff
@@ -226,6 +227,19 @@ async fn filesystem_put(
     Ok(HttpResponseUpdatedNoContent())
 }
 
+/// Initializes a CockroachDB cluster
+#[endpoint {
+    method = POST,
+    path = "/cockroachdb",
+}]
+async fn cockroachdb_init(
+    rqctx: RequestContext<SledAgent>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let sa = rqctx.context();
+    sa.cockroachdb_initialize().await?;
+    Ok(HttpResponseUpdatedNoContent())
+}
+
```

Comment:
Agreed with your comment that "this is sketchy, and it would be cool to not have it".

It seems like it could also be possible to have the method script auto-initialize the DB if it's empty, but that seems like it could raise other problems.

All that said, this approach is probably good enough to last a while.